RELEASE.md

```diff
@@ -12,25 +12,37 @@ be cut at the same time.
 
 The process for both is as follows:
 
-1. An issue is proposing a new release with a changelog since the last release
-1. At least two of the [OWNERS](OWNERS) must LGTM this release
+1. An issue is proposing a new release with a changelog since the last release.
+   The issue should describe if one or both of the container image and Helm
+   chart should be released.
+     - Example issue: https://github.com/kubernetes-sigs/dra-example-driver/issues/119
+1. At least two of the [OWNERS](OWNERS) must agree to move forward with this
+   release.
 1. When releasing new container images, the Helm chart's `appVersion` in
    Chart.yaml is updated to reflect the version of the images to be cut.
+    - Example PR: https://github.com/kubernetes-sigs/dra-example-driver/pull/120
 1. Depending on what is being released, an OWNER runs `git tag -a -s $VERSION`
-   and inserts the changelog and pushes the tag with `git push $VERSION`
+   and inserts the changelog and pushes the tag with `git push <remote> $VERSION`
     - When releasing the container images, `$VERSION` is a `v`-prefixed
       [SemVer], e.g. `v0.1.0`
     - When releasing the Helm chart, `$VERSION` is a `chart/`-prefixed [SemVer],
       e.g. `chart/0.1.0`
     - The same commit may be tagged with a tag of each form to release both the
       Helm chart and container images at the same time
+    - `<remote>` refers to the name of the Git remote for the repo at
+      [https://github.com/kubernetes-sigs/dra-example-driver], likely either
+      `upstream` or `origin`
     - Each pushed tag triggers builds of artifacts pushed to the staging
       repository for the [dra-example-driver container image][container-staging]
       and [Helm chart][chart-staging]
-1. A PR updating the [registry.k8s.io image list][image-list] is opened with the
+1. A PR updating the [registry.k8s.io image list][image list] is opened with the
    SHA digests and tags of the new artifacts from the staging repo
     - Example PR: https://github.com/kubernetes/k8s.io/pull/7871
-1. The artifacts are verified to be available from registry.k8s.io
+1. The artifacts are verified to be available from registry.k8s.io:
```

Comment:
Link to the [image promotion job](https://prow.k8s.io/job-history/gs/kubernetes-ci-logs/logs/post-k8sio-image-promo) would be also nice to have, so we'd know where to look if images are not promoted.